source/extensions/filters/network/mongo_proxy/codec_impl.cc

```diff
@@ -222,6 +222,119 @@ std::string ReplyMessageImpl::toString(bool full) const {
       full ? documentListToString(documents_) : std::to_string(documents_.size()));
 }
 
+/*
+ * OP_COMMAND mongo message implementation
+ */
+void CommandMessageImpl::fromBuffer(uint32_t message_length, Buffer::Instance& data) {
+  ENVOY_LOG(trace, "decoding COMMAND message");
+  uint64_t original_data_length = data.length();
+
+  database_ = Bson::BufferHelper::removeCString(data);
+  commandName_ = Bson::BufferHelper::removeCString(data);
+  metadata_ = Bson::DocumentImpl::create(data);
+  commandArgs_ = Bson::DocumentImpl::create(data);
+
+  // There may be additional docs
+  if (data.length() - (original_data_length - message_length) > 0) {
+    inputDocs_.emplace_back(Bson::DocumentImpl::create(data));
+  }
+
+  ENVOY_LOG(trace, "{}", toString(true));
+}
+
+std::string CommandMessageImpl::toString(bool full) const {
+  return fmt::format(
+      R"EOF({{"opcode": "OP_COMMAND", "id": {}, "response_to": {}, "database": "{}", )EOF"
+      R"EOF("commandName": "{}", "metadata": {}, )EOF"
+      R"EOF("commandArgs": {}, "inputDocs": {}}})EOF",
+      request_id_, response_to_, database_.c_str(), commandName_.c_str(), metadata_->toString(),
+      commandArgs_->toString(),
+      full ? documentListToString(inputDocs_) : std::to_string(inputDocs_.size()));
+}
+
+bool CommandMessageImpl::operator==(const CommandMessage& rhs) const {
+  if (!(requestId() == rhs.requestId() && responseTo() == rhs.responseTo() &&
+        database_ == rhs.database() && commandName_ == rhs.commandName() &&
```

Comment:
use the accessor instead of the underlying variable

Reply:
I believe that calling accessor on "this" pointer is unnecessary method call, but I changed it as you suggested. For all other classes, == operator is implemented that way, so using accessor brings some consistency.